### PR TITLE
Remove the GetLeavesByIndex method from the log storage interface.

### PR DESCRIPTION
**Pull request by pphaneuf**
Part of #2245.

### CODE REVIEW

File: storage/mysql/log_storage_test.go [hunk 1]
```diff
@@ -64,8 +61,7 @@ var fakeDequeueCutoffTime = time.Date(2016, 11, 10, 15, 16, 30, 0, time.UTC)
 
 // Used for tests involving extra data
 var (
-	someExtraData  = []byte("Some extra data")
-	someExtraData2 = []byte("Some even more extra data")
+	someExtraData = []byte("Some extra data")
 )
```
Comment (pavelkalinnikov) [inline storage/mysql/log_storage_test.go:65]: Join these as one line?

File: storage/tools/dump_tree/dumplib.go [hunk 2]
```diff
@@ -474,9 +474,9 @@ func dumpLeaves(ctx context.Context, ls storage.LogStorage, tree *trillian.Tree,
 	}()
 
 	for l := int64(0); l < int64(ts); l++ {
```
Comment (pavelkalinnikov) [inline storage/tools/dump_tree/dumplib.go:476]: Consider squashing this to one `GetLeavesByRange` call (in this or separate PR).

**Review by pavelkalinnikov — APPROVED** [review]

File: storage/tools/dump_tree/dumplib.go (hunk 2 above)
Comment (pphaneuf) [reply storage/tools/dump_tree/dumplib.go:475]: I'll do this in a separate PR.

File: storage/mysql/log_storage_test.go (hunk 1 above)
Comment (pphaneuf) [reply storage/mysql/log_storage_test.go:65]: Done.

File: storage/tools/dump_tree/dumplib.go (hunk 2 above)
Comment (pphaneuf) [reply storage/tools/dump_tree/dumplib.go:475]: I'll do this in a separate PR.

**Pull request merged**